src_c/camera_v4l2.c

```diff
@@ -57,13 +75,9 @@ v4l2_list_cameras(int *num_devices)
         close(fd);
```

Comment:
Suggestion: this code block
```

        fd = open(device, O_RDONLY);
        if (fd != -1) {
            devices[num] = device;
            num++;
            device = (char *)malloc(sizeof(char) * 13);
        }
        close(fd);
```
is on line 49 already, maybe it could be extracted to its own method?

Reply:
Thanks for the review! 
Hmmm however I don't think this would work well as a seperate function, because there are many local variables it needs to make use of, like `num`, `device` and `devices` and passing all of them by ref as args seems excessive for a 7 line code block to me. It would work better as a macro, but again, macros must be avoided when they can.

Reply:
Now that you reviewed on this bit of code, I see the malloc return value is not been error-checked here, which needs to be done